docker_mcp/services/stack/migration_orchestrator.py

```
@@ -0,0 +1,359 @@
+        self.network = StackNetwork()
+        self.risk_assessment = StackRiskAssessment()
+        self.volume_utils = StackVolumeUtils()
+        self.executor = StackMigrationExecutor(config, context_manager)
+
+    def _validate_host(self, host_id: str) -> tuple[bool, str]:
+        """Validate host exists in configuration."""
+        if host_id not in self.config.hosts:
+            return False, f"Host '{host_id}' not found"
+        return True, ""
+
+    async def migrate_stack(
```

Ruff (0.12.2)

`migrate_stack` is too complex (25 > 10)

(C901)